iguana.commons/src/main/java/org/aksw/iguana/commons/io/BigByteArrayOutputStream.java

```diff
@@ -0,0 +1,108 @@
+package org.aksw.iguana.commons.io;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BigByteArrayOutputStream extends OutputStream {
+
+    private List<ByteArrayOutputStream> baos = new ArrayList<ByteArrayOutputStream>();
+
+    public BigByteArrayOutputStream() {
+        baos.add(new ByteArrayOutputStream());
+    }
+
+
+    public List<ByteArrayOutputStream> getBaos() {
+        return baos;
+    }
+
+    public void write(BigByteArrayOutputStream bbaos) throws IOException {
+        for (byte[] bao : bbaos.toByteArray()) {
+            for (Byte b : bao) {
+                write(b);
+            }
+        }
+
+    }
+
+    public long size() {
+        long ret = 0;
+        for (ByteArrayOutputStream ba : baos) {
+            ret += ba.size();
+        }
+        return ret;
+    }
+
+    public synchronized byte[][] toByteArray() {
+        byte[][] ret = new byte[baos.size()][];
+        for (int i = 0; i < baos.size(); i++) {
+            ret[i] = baos.get(i).toByteArray();
+        }
+        return ret;
+    }
+
+
+    public void write(byte[] i) throws IOException {
+        for (byte b : i) {
+            write(b);
+        }
+    }
+
+    public void write(byte[][] i) throws IOException {
+        for (byte[] arr : i) {
+            for (byte b : arr) {
+                write(b);
+            }
+        }
+    }
+
+    public void write(byte i) throws IOException {
+        ByteArrayOutputStream current = baos.get(baos.size() - 1);
+        current = ensureSpace(current);
+        current.write(i);
+    }
+
+    @Override
+    public void write(int i) throws IOException {
+        ByteArrayOutputStream current = baos.get(baos.size() - 1);
+        current = ensureSpace(current);
+        current.write(i);
+    }
+
+    private ByteArrayOutputStream ensureSpace(ByteArrayOutputStream current) {
+        if (current.size() == 2147483639) {
```

Comment:
What's that number?
I would have expected
2147483645 = Integer.MAX_VALUE - 2 not
2147483639